src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java

```diff
@@ -158,7 +171,8 @@ private UacQidLink createNewUacQidPair(Case linkedCase, String questionnaireType
 
   private boolean isQuestionnaireWelsh(String treatmentCode) {
     return (treatmentCode.startsWith(HOUSEHOLD_INITIAL_CONTACT_QUESTIONNAIRE_TREATMENT_CODE_PREFIX)
-        && treatmentCode.endsWith(WALES_TREATMENT_CODE_SUFFIX));
+        || treatmentCode.startsWith(CE_INITIAL_CONTACT_QUESTIONNAIRE_TREATMENT_CODE_PREFIX)
```

Comment:
I have a concern that the boolean logic of this is slightly hard to work out:

`IF treatment code starts HH_Q OR treatment code starts CE_Q AND treatment code ends W THEN...`

It's not clear whether this always makes sure that the treatment code ends W. You should put the first part of the clause into its own clause.

Suggestion:

```
return ((treatmentCode.startsWith(HOUSEHOLD_INITIAL_CONTACT_QUESTIONNAIRE_TREATMENT_CODE_PREFIX)
        || treatmentCode.startsWith(CE_INITIAL_CONTACT_QUESTIONNAIRE_TREATMENT_CODE_PREFIX))
        && treatmentCode.endsWith(WALES_TREATMENT_CODE_SUFFIX));
```

Reply:
Done